polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1783,39 +1807,37 @@ async fn kick_off_seconding<Context>(
 		},
 	};
 	let collations = &mut per_relay_parent.collations;
-	let relay_parent_mode = per_relay_parent.prospective_parachains_mode;
 
 	let fetched_collation = FetchedCollation::from(&candidate_receipt);
 	if let Entry::Vacant(entry) = state.fetched_candidates.entry(fetched_collation) {
 		collation_event.pending_collation.commitments_hash =
 			Some(candidate_receipt.commitments_hash);
 
-		let pvd =
-			match (relay_parent_mode, collation_event.pending_collation.prospective_candidate) {
-				(
-					ProspectiveParachainsMode::Enabled { .. },
-					Some(ProspectiveCandidate { parent_head_data_hash, .. }),
-				) =>
-					request_prospective_validation_data(
-						ctx.sender(),
-						relay_parent,
-						parent_head_data_hash,
-						pending_collation.para_id,
-					)
-					.await?,
-				(ProspectiveParachainsMode::Disabled, _) =>
-					request_persisted_validation_data(
-						ctx.sender(),
-						candidate_receipt.descriptor().relay_parent,
-						candidate_receipt.descriptor().para_id,
-					)
-					.await?,
-				_ => {
-					// `handle_advertisement` checks for protocol mismatch.
-					return Ok(())
-				},
-			}
-			.ok_or(SecondingError::PersistedValidationDataNotFound)?;
+		let pvd = match (
+			collation_event.collator_protocol_version,
+			collation_event.pending_collation.prospective_candidate,
+		) {
+			(CollationVersion::V2, Some(ProspectiveCandidate { parent_head_data_hash, .. })) =>
+				request_prospective_validation_data(
+					ctx.sender(),
+					relay_parent,
+					parent_head_data_hash,
+					pending_collation.para_id,
+				)
+				.await?,
+			(CollationVersion::V1, _) =>
+				request_persisted_validation_data(
+					ctx.sender(),
+					candidate_receipt.descriptor().relay_parent,
+					candidate_receipt.descriptor().para_id,
+				)
+				.await?,
```

Comment:
Previously we would've requested pvd from chain even for V2 protocol in case async backing is disabled.
This is correct, because prospective parachains subsystem doesn't do any job otherwise.

With new version this is not the case anymore, and here it implies that V2 -> async backing is enabled, which is wrong assumption.

So it should be
```rust
(CollationVersion::V2, Some(ProspectiveCandidate { parent_head_data_hash, .. })) 
    if relay_parent_mode.is_enabled() => // <-- this
				request_prospective_validation_data(
					ctx.sender(),
					relay_parent,
					parent_head_data_hash,
					pending_collation.para_id,
				)
				.await?,
```